### PR TITLE
feat(replays): Create a Share button to allow sharing replays at a specific timestamp

### LINKED ISSUE #41731: Share at timestamp
### Problem
Interesting moments in a replay can last a few seconds but our replays can be much longer. Meaning if an engineer wanted to share that interesting moment with their colleagues, they would have to give them precise times to skim to. That sucks

### Acceptance Critertia
- User can share a Replay's URL which starts from the beginning
- User can share a Replay's URL which starts at a specific point in time
- If the user has already played a replay up to a specific point in time, that time is prefilled in the "Start at" field

### Designs
[Figma Designs](https://www.figma.com/file/VTwvF65Dvg710T7k2Jvghl/Specs%3A-Share-Modal?node-id=0%3A1&t=VThObqaDgowc3uZL-1)

Image
Image
Image

### PR DESCRIPTION
**Pull request by ryan953**
Here's the modal with a custom timestamp that someone wants to share:

SCR-20221129-dxz

By default the timestamp is the current time of the player, in this example 2 minutes into the replay:
SCR-20221129-dxt

1. When the modal opens it starts with the default time, `2:00` in this case.
2. Check the checkbox and edit the time: it'll reflect in the url right away
3. If you un-check the checkmark, so it's not going to use a custom time anymore, then the text box will go back to saying the default `2:00` again.
4. Click the checkmark to bring back the custom time you entered in step 2

Fixes #41731

### CODE REVIEW

**Comment by Jesse-Box** [comment]
@ryan953 the inputs on this modal vary in heights, I'm gonna assume the copy to clipboard has some overrides to make it shorter, but in this specific context its better to stick with the default height. Also the width of the `Start at` input is very wide, can we trim the width a little?

File: static/app/components/inputGroup.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const InputGroupContext = createContext<InputContext>({inputProps: {}});
  *     <InputTrailingItems> … </InputTrailingItems>
  *   </InputGroup>
  */
-export function InputGroup({children}: React.HTMLAttributes<HTMLDivElement>) {
+export function InputGroup({children, className}: React.HTMLAttributes<HTMLDivElement>) {
```
Comment (ryan953) [inline static/app/components/inputGroup.tsx:47]: It looks like this was meant to work earlier, but was missed: https://github.com/getsentry/sentry/blob/master/static/app/components/textCopyInput.tsx#L93

No other callsites for `<InputGroup>` pass in a `className`, so i'm not worried about some styles getting ressurected someplace. One site comes close, ruleNode.tsx styles a `<NumberInput>` but that skips styling the group and styles its inner input directly.

**Comment by ryan953** [comment]
@Jesse-Box updated the input box sizes (height and width). New screen shots in the description for reference.

File: static/app/utils/formatters.spec.tsx [hunk 2]
```diff
@@ -77,6 +82,78 @@ describe('getDuration()', function () {
   });
 });
 
+describe('formatSecondsToClock', function () {
+  it('should format durations', function () {
+    expect(formatSecondsToClock(0)).toBe('00:00');
```
Comment (ryan953) [inline static/app/utils/formatters.spec.tsx:87]:
```suggestion
    expect(formatSecondsToClock(0)).toBe('0:00');
```

File: static/app/utils/formatters.spec.tsx [hunk 3]
```diff
@@ -77,6 +82,78 @@ describe('getDuration()', function () {
...
+    expect(formatSecondsToClock(65.123)).toBe('01:05.123');
+    expect(formatSecondsToClock(122)).toBe('02:02');
+    expect(formatSecondsToClock(3720)).toBe('01:02:00');
+    expect(formatSecondsToClock(36000)).toBe('10:00:00');
+    expect(formatSecondsToClock(86400)).toBe('24:00:00');
+    expect(formatSecondsToClock(86400 * 2)).toBe('48:00:00');
+  });
+
+  it('should format negative durations', function () {
+    expect(formatSecondsToClock(-0)).toBe('00:00');
```
Comment (ryan953) [inline static/app/utils/formatters.spec.tsx:101]:
```suggestion
    expect(formatSecondsToClock(-0)).toBe('0:00');
```

**Review by Jesse-Box — APPROVED** [review]

**Comment by Jesse-Box** [comment]
> @Jesse-Box updated the input box sizes (height and width). New screen shots in the description for reference.

looks great man.

**Pull request merged**